### PR TITLE
Pu/vectorization

### PR DESCRIPTION
**Pull request by izulin**
### Context

Vectorization of scalar functions.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1.
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,8 +22,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 - **Breaking change**: Changed API of many sheet-related methods to take sheetId instead of sheetName as an argument. (#645)
 - **Breaking change**: Removed support for matrix formulas (`{=FORMULA}`) notation. Engine now supports formulas returning array of values (instead of only scalars). (#652)
-- Changed SWITCH function so it takes array as its first argument.
-- **Breaking change**: Removed numeric matrix detection along with matrixDetection and matrixDetectionThreshold config options. (#669)
+- **Breaking change**: Removed numeric matrix detection along with matrixDetection and matrixDetectionThreshold config options. (#669)
```
Comment (izulin) [inline CHANGELOG.md:25]:
```suggestion
- Changed SWITCH function so it takes array as its first argument.
- **Breaking change**: Removed numeric matrix detection along with matrixDetection and matrixDetectionThreshold config options. (#669)
```

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

File: src/interpreter/plugin/FunctionPlugin.ts [hunk 2]
```diff
@@ -54,6 +54,11 @@ export interface FunctionArguments {
    * Return number value is packed into this subtype.
    */
   returnNumberType?: NumberType,
+
+  /**
+   * Some function do not allow vectorization: array-output, and special functions.
+   */
+  vectorizationForbidden?: boolean,
```
[the remark this answers is not in the document]
Comment (izulin) [reply src/interpreter/plugin/FunctionPlugin.ts:61]: There are subtle differences between parameters that are used by the core engine, and parameters that are just a guiding info for our wrappers.
That is, unless the custom function is written in a manner that supports vectorization, NOT setting `vectorizationForbidden` will not produce valid result anyway.

[the remark this answers is not in the document]
Comment (wojciechczerniak) [reply src/interpreter/plugin/FunctionPlugin.ts:61]: And yet, all those parameters may be used for custom function, right?

[the remark this answers is not in the document]
Comment (izulin) [reply src/interpreter/plugin/FunctionPlugin.ts:61]: Yes, they are all passed. But some of those have no effect unless explicitly used by the function implementation.

**Review by wojciechczerniak — APPROVED** [review]
👍

**Pull request merged**